client/hooks/use-teams.ts

```diff
@@ -0,0 +1,68 @@
+import { ipAtom } from '@/lib/services/api';
+import { convertTeam, getTeams, RawTeamInfo, TeamInfo } from '@/lib/services/teams';
+import { useWebSocket } from '@/lib/services/ws';
+import { useQuery } from '@tanstack/react-query';
+import { atom, useAtom, useAtomValue } from 'jotai';
+
+const teamsAtom = atom<Record<string, TeamInfo>>({});
+const teamsListAtom = atom((get) => Object.values(get(teamsAtom)));
+const selectedTeamAtom = atom<TeamInfo | null>(null);
+export const useTeams = () => {
+    const ip = useAtomValue(ipAtom);
+    const { ws: basaltWs } = useWebSocket();
+    const [teams, setTeams] = useAtom(teamsAtom);
+    const teamsList = useAtomValue(teamsListAtom);
+    const [selectedTeam, setSelectedTeam] = useAtom(selectedTeamAtom);
+
+    const updateTeams = (teams: TeamInfo[]) => {
+        setSelectedTeam((prev) => teams.find((t) => t.team === prev?.team) ?? null);
+        setTeams(teams.reduce((acc, t) => ({ ...acc, [t.team]: t }), {}));
+    };
+    const updateTeam = (rawTeam: RawTeamInfo) => {
+        const parsedTeam = convertTeam(rawTeam);
+        setSelectedTeam((prev) =>
+            prev ? (prev.team === parsedTeam.team ? parsedTeam : prev) : prev
+        );
+        setTeams((prev) => {
+            prev[parsedTeam.team] = parsedTeam;
+            // has to be new object so jotai actually knows it changed... (L)
+            return { ...prev };
+        });
```

Comment:
If you wanted this to be more pure (and cleaner):
```suggestion
        setTeams((prev) => ({
            ...prev,
            [parsedTeam.team]: parsedTeam,
        }));
```

Reply:
Good idea. Updating